src/constants.js

```diff
@@ -1,4 +1,13 @@
-export const scaleFactor = 2;
+// Actual scale factor based on screen size
+const header = document.getElementById('header');
+const leftPanel = document.getElementById('left-panel');
+const canvas = document.getElementById('game');
+const footer = document.getElementById('footer');
+
+const scaleFactorWidth = (canvas.offsetWidth - leftPanel.offsetWidth * 2) / 480;
+const scaleFactorHeight =
+    (canvas.offsetHeight - (header.offsetHeight + footer.offsetHeight)) / 320;
+export const scaleFactor = Math.min(scaleFactorWidth, scaleFactorHeight);
```

Comment:
This doesn't work on desktop nor on mobile. Only a scale factor of 2 looks good what I figured out.
If we calculate the scale factor here, then everything will be to small globally.
Maybe you can set the scale factor for the mini game only once it started and switch back to the default one after exiting the fishing game?